lib/index.js

```diff
@@ -1,30 +1,39 @@
 'use strict';
 
 const Fs = require('fs');
-const Path = require('path');
 const Util = require('util');
+const Glob = require('glob');
 
 const internals = {
-    readDir: Util.promisify(Fs.readdir),
-    testDefault: /\.(js)$/
+    testDefault: /\.(js)$/,
+    glob: Util.promisify(Glob),
+    checkPath: (path) => {
+
+        const chunk = path.replace(/[\/\*]+$/, '');
+        return Util.promisify(Fs.exists)(chunk);
+    }
 };
 
 module.exports = {
     pkg: require('../package.json'),
     register: async (server, options) => {
 
-        const fileList = await internals.readDir(options.dir);
         const fileTest = options.test || internals.testDefault;
 
-        fileList.forEach((file) => {
+        if (await internals.checkPath(options.dir)) {
```

Comment:
I don't think this is check is needed, or correct.
If you test `src/**/routes/**/*.js` it will fail if I'm correct.
If you want to error just error when glob returns an empty array.

Reply:
@AdriVanHoudt Simple enough. Will do.